pyispyb/core/modules/proposals.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional
+from sqlalchemy.orm import joinedload, contains_eager
+from ispyb import models
+from pyispyb.app.extensions.database.utils import Paged, page
+from pyispyb.app.extensions.database.middleware import db
+
+
+def get_proposals(
+    skip: int,
+    limit: int,
+    proposalId: Optional[int] = None,
+    proposalCode: Optional[str] = None,
+    proposalNumber: Optional[str] = None,
+    proposalHasPerson: Optional[bool] = False,
+) -> Paged[models.Proposal]:
+
+    query = db.session.query(models.Proposal).options(
+        joinedload(models.Proposal.Person)
+    )
+
+    if proposalHasPerson:
+        query = (
+            query.outerjoin(
+                models.ProposalHasPerson,
+                models.Proposal.proposalId == models.ProposalHasPerson.proposalId,
+            )
+            .options(contains_eager("ProposalHasPerson"))
+            .outerjoin(
+                models.Person,
+                models.ProposalHasPerson.personId == models.Person.personId,
+            )
+            .options(contains_eager("ProposalHasPerson.Person"))
+        )
+
+    if proposalId:
+        query = query.filter(models.Proposal.proposalId == proposalId)
+
+    if proposalCode and proposalNumber:
+        query = query.filter(models.Proposal.proposalCode == proposalCode)
+        query = query.filter(models.Proposal.proposalNumber == proposalNumber)
+
+    # https://github.com/aiidateam/aiida-core/issues/1600
+    query_distinct = query.distinct()
```

Comment:
This is odd, if you try `query.group_by(models.Proposal.proposalId)` do you get the same result? Probably worth `print(query)` to really see what is going on here.

Reply:
I fixed it by using distinct on the query joining the proposalHasPerson table